### PR TITLE
WIP Remove Int timestep variant

### PR DESCRIPTION
**Pull request by lrennels**
See issue #247

### CODE REVIEW

File: src/core/types.jl [hunk 1]
```diff
@@ -327,7 +327,7 @@ mutable struct ComponentInstance{TV <: ComponentInstanceVariables, TP <: Compone
 
     run_timestep::Union{Void, Function}
     use_integer_time::Bool                  # will be deprecated when switching to timesteps in all cases
```
Comment (rjplevin) [inline src/core/types.jl]: Comment out `use_integer_time::Bool`, too.

**Comment by lrennels** [comment]
If we want `t` to be automatically assumed to be an `AbstractTimestep` within `run_timestep`, then we need a few changes, some to the API so I want to check in with @davidanthoff and @rjplevin sine these are substantial changes and we may want to think of ways to simplify the API as this introduces opportunities for bugs especially with needing `gettime(ts)` and `ts.t`.  You can see the changes below in the comments.

```
@defcomp Bar2 begin
    inputB = Parameter(index=[time])
    output = Variable(index=[time])
    
    function run_timestep(p, v, d, ts) #ts will be assumed to be an AbstractTimestep
        if gettime(ts) < 2005 #we use gettime(ts) here and interpret it as a year not an index
            v.output[ts] = 0
        else
            v.output[ts] = p.inputB[ts.t-1] * ts.t #we have to use ts.t here
        end
    end
end
```

**Comment by rjplevin** [comment]
As I recall, we discussed adding methods for arithmetic operators to Timestep, which would solve the `ts - 1` problem. I'm less sure what to do about `x * ts`. Is that used in real models, or only in testing?

Our `getindex` methods should handle the case of `p.inputB[ts]`. @davidanthoff - is this consistent with how you were expecting this to work?

**Comment by lrennels** [comment]
Yep, adding arithmetic operators would fix the `ts - 1`.  I started to add those but then went down a bit of a rabbit hole with the cases such as do we want users to have to do `gettime(ts) == 2005` and the `x * t` case so I paused.

**Comment by lrennels** [comment]
Got it, yes I too wonder if that's a real use case, along with something like *x + ts* or *x - ts* ... which would seem silly to me.  It seems that if you wanted to do those you also could just create your own counter variable to use in their place ... but I guess it depends on if these use cases are common at all.  I've never seen them.

**Comment by rjplevin** [comment]
I think this is ready to go once you address merge conflicts. I think once we get all the changes related to Int timesteps in we'll stop having CI errors.

**Pull request merged**